app/services/rate_checker.rb

```diff
@@ -1,7 +1,20 @@
 class RateChecker
-  def user_has_already_rated?(submission_id, user_id)
-    submission = Submission.find(submission_id)
+  def initialize(submission_id, user_id)
```

Comment:
After the conversation about DI I'd like to inject ready objects in the constructor and parameters to `build` methods, what do you think?